### PR TITLE
applications: nrf5340_audio: Reimplement function for calculating err_us

### PR DESCRIPTION
**Pull request by alexsven**
- Fix bug where remainder operations were done on negative numbers
- Create helper function to fix both uses in one place

### CODE REVIEW

File: applications/nrf5340_audio/src/audio/audio_datapath.c [hunk 1]
```diff
@@ -160,6 +160,42 @@ static bool tone_active;
...
+ * @brief	Calculate error between sdu_ref and frame_start_ts.
+ *
+ * @note	Used to adjust audio clock to account for drift.
+ *
+ * @param	sdu_ref_us	Timestamp for sdu.
+ * @param	frame_start_ts	Timestamp for I2S.
+ *
+ * @return	err_us	Error in microseconds.
+ */
+static int32_t err_us_calculate(uint32_t sdu_ref_us, uint32_t frame_start_ts)
```
Comment (koffes) [inline applications/nrf5340_audio/src/audio/audio_datapath.c:173]:
```suggestion
static int32_t err_calculate_us(uint32_t sdu_ref_us, uint32_t frame_start_ts)
```
Nitpick. Discuss

File: applications/nrf5340_audio/src/audio/audio_datapath.c [hunk 2]
```diff
@@ -160,6 +160,42 @@ static bool tone_active;
...
+{
+	int32_t total_err = (sdu_ref_us - frame_start_ts);
```
Comment (koffes) [inline applications/nrf5340_audio/src/audio/audio_datapath.c:175]: Must be int64_t

File: applications/nrf5340_audio/src/audio/audio_datapath.c [hunk 3]
```diff
@@ -160,6 +160,42 @@ static bool tone_active;
...
+
+	bool err_neg = false;
```
Comment (koffes) [inline applications/nrf5340_audio/src/audio/audio_datapath.c:177]: Move to top of function.

File: applications/nrf5340_audio/src/audio/audio_datapath.c [hunk 4]
```diff
@@ -160,6 +160,42 @@ static bool tone_active;
...
+
+	/* Store sign for later use, since modulo operation is undefined for negatives */
+	if (total_err < 0) {
+		err_neg = true;
+		total_err *= -1;
+	}
+
```
Comment (koffes) [inline applications/nrf5340_audio/src/audio/audio_datapath.c:184]: Add comment: How we later take errors over 1000 us into account.

**Review by koffes — CHANGES_REQUESTED** [review]

**Review by koffes — APPROVED** [review]

**Review by gWacey — APPROVED** [review]

**Comment by rick1082** [comment]
> We need to figure out why we get more negative values for ctrl_blk.previous_sdu_ref_us - frame_start_ts using SDC

Pardon me a dumb question:
Would this because of the different definition between PCFT and SDC regarding with the sdu_ref?

**Pull request merged**